strings-csharp/GetFirstNCharactersOfAString/GetFirstNCharactersOfAString/FirstNCharactersOfStringGetter.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System.Text;
+
+namespace GetFirstNCharactersOfAString;
+public class FirstNCharactersOfStringGetter(string inputString = "CodeMaze", int numberOfCharacters = 4)
+{
+    private readonly string _inputString = inputString;
+    private readonly int _numberOfCharacters = numberOfCharacters;
+
+    private bool IsInputStringShorter()
+        => _inputString.Length < _numberOfCharacters;
+
+    public ReadOnlySpan<char> UseForLoop()
+    {
+        if (IsInputStringShorter())
+            return _inputString;
+
+        var strBuilder = new StringBuilder();
```

Comment:
This won't make a huge difference because this method will still be one of the slowest, but we should at least use the `StringBuilder` overload that takes an initial capacity argument. We already know how long our final string will be, so this will reduce reallocations in the StringBuilder